ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -0,0 +1,46 @@
+(ns cmr.ingest.api.subscriptions
+  "Subscription ingest functions in support of the ingest API."
+  (:require
+   [cmr.acl.core :as acl]
+   [cmr.common-app.api.enabled :as common-enabled]
+   [cmr.common-app.api.launchpad-token-validation :as lt-validation]
+   [cmr.common.log :refer [debug info warn error]]
+   [cmr.ingest.api.core :as api-core]
+   [cmr.ingest.services.ingest-service :as ingest]
+   [cmr.ingest.validation.validation :as v]))
+
+(defn- validate-and-prepare-subscription-concept
+  "Validate subscription concept, set the concept format and returns the concept;
+  throws error if the metadata is not a valid against the UMM subscription JSON schema."
+  [concept]
+  (let [concept (update-in concept [:format] (partial ingest/fix-ingest-concept-format :subscription))]
+    (v/validate-concept-request concept)
+    (v/validate-concept-metadata concept)
+    concept))
+
+(defn ingest-subscription
+  "Processes a request to create or update a subscription."
+  [provider-id native-id request]
+  (let [{:keys [body content-type headers request-context]} request
+        concept (api-core/body->concept!
+                 :subscription provider-id native-id body content-type headers)]
+    (lt-validation/validate-launchpad-token request-context)
+    (api-core/verify-provider-exists request-context provider-id)
+    (acl/verify-ingest-management-permission
+      request-context :update :provider-object provider-id)
+    (common-enabled/validate-write-enabled request-context "ingest")
+    (let [concept (validate-and-prepare-subscription-concept concept)
```

Comment:
This function shouldn't have two `let` statements in it. Is there a way you can break it up into smaller chunks?

Reply:
Done.